modules/axhal/src/arch/aarch64/trap.S

```diff
@@ -21,6 +21,11 @@
     mrs     x11, spsr_el1
     stp     x30, x9, [sp, 30 * 8]
     stp     x10, x11, [sp, 32 * 8]
+    
+    # We may have interrupted userspace, or a guest, or exit-from or 
+    # return-to either of those. So we can't trust sp_el0, and need to
+    # restore it.
+    bl      {cache_current_task_ptr}
```

Comment:
you can add `sp_el0` explicitl in thic function name, to claim that we use `sp_el0` to store current task ptr
BTW, is it a caching operation??

Reply:
In ArceOS's aarch64 architecture, we use `sp_el0` as the read cache for the current task pointer. And this function `cache_current_task_ptr` will update this cache.